doc/release-notes.md

```diff
@@ -75,19 +75,25 @@ There is a checkbox to mark the transaction as final.
 The RPC default remains unchanged: to use RBF, launch with `-walletrbf=1` or
 use the `replaceable` argument for individual transactions.
 
-Custom wallet directories
----------------------
-The ability to specify a directory other than the default data directory in which to store
-wallets has been added. An existing directory can be specified using the `-walletdir=<dir>`
-argument. Wallets loaded via `-wallet` arguments must be in this wallet directory. Care should be taken
-when choosing a wallet directory location, as if it becomes unavailable during operation,
-funds may be lost.
-
-Default wallet directory change
---------------------------
-On new installations (if the data directory doesn't exist), wallets will now be stored in a
-new `wallets/` subdirectory inside the data directory. If this `wallets/` subdirectory
-doesn't exist (i.e. on existing nodes), the current datadir root is used instead, as it was.
+Wallets directory configuration (`-walletdir`)
+----------------------------------------------
+
+Bitcoin Core now has more flexibility in where the wallets directory can be located.
+
+- For new installations (where the data directory doesn't already exist),
+  wallets will now be stored in a new `wallets/` subdirectory inside the data
+  directory by default.
+- For existing nodes (where the data directory already exists), wallets will be
+  stored in the data directory root by default. If a `wallets/` subdirectory
+  already exists in the data directory root, then wallets will be stored in the
+  `wallets/` subdirectory by default.
+- The location of the wallets directory can be overridden by specifying a
+  `-walletdir=<path>` option where `<path>` can be an absolute path or a
+  relative path (relative to the current working directory). `<path>` can
```

Comment:
Relative to the data directory, not the current working directory, I believe.

Reply:
No, I've retested and it's relative to the cwd. See https://github.com/bitcoin/bitcoin/blob/44080a90a29292df96e92f22242785c5040000a1/src/wallet/walletutil.cpp#L12.

This is part of what prompted my comment here: https://github.com/bitcoin/bitcoin/pull/11687#issuecomment-356136301 - it's slightly  confusing that `-walletdir` relative paths are relative to the cwd and `-wallet` paths are relative to the `walletdir`. That's also why I think #11687 can't be thought of as being completely independent from the changes in #11466 . The documentation needs to describe clearly the interactions between the two options.

Reply:
Attn @MeshCollider 

> it's slightly confusing that -walletdir relative paths are relative to the cwd

This is worse than confusing. This sounds like a mistake that should be fixed. Current working directory is helpful thing to rely on in scripts and command line tools, but long running apps and daemons should not depend on the current working directory. If a relative -walletdir path is specified in bitcoin.conf, it would be footgun for that directory to point to a completely different location if one day the user happens to be in a different directory when starting up bitcoin.

I think a new PR should be opened to either forbid relative -walletdir paths, or interpret walletdir paths stably relative to `<datadir>`.

Reply:
The -walletdir is relative in the same way -datadir is, I think that consistency is less confusing than making walletdir relative to datadir. Keep in mind an error will be thrown if the wallet directory doesn't exist so it's unlikely to start if the cwd was different

Reply:
I don't have a strong preference on how this works. As @MeshCollider says, the relativity of `-walletdir` is consistent with `-datadir`.

Any approach (make walletdir relative to datadir, make walletdir relative to cwd, or disallow relative paths for walletdir) seems fine to me as long as it's clearly documented.

Reply:
I've noticed, on occasion, that people ask "Where is my wallet.dat file stored?"  I wonder if this "cwd" default behaviour might be a culprit.  I certainly favour clearer documentation, and I wonder if perhaps it could be helpful to raise a notice indicating the full path when a new wallet file is created.

Reply:
@randolf this is unrelated, this change has not been in any release yet, it will be released in 0.16 :)